### PR TITLE
Parse combined AS_QUALapprox values from older reblocked GVCFs properly

### PR DESCRIPTION
**Pull request by ldgauthier**
Sometimes NON_REF gets a zero and sometimes it's empty.  This seems isolated to a much older version of ReblockGVCF, but that was what we were running for production pipeline tests.

@droazen I'd like this to go into this week's release

### CODE REVIEW

**Comment by ldgauthier** [comment]
@droazen this fix is the same as a bug you caught in another review -- https://github.com/broadinstitute/gatk/pull/6079#discussion_r332183572  Should be an easy review?

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/allelespecific/AS_QualByDepthUnitTest.java [hunk 1]
```diff
@@ -27,4 +34,16 @@ protected String getKey() {
...
 
+    @Test
+    public void testParseQualList() {
+        final String goodQualList = "|234|0"; //combined VCs from GenomicsDB have zero value for NON-REF
+        final String trickyQualList = "|234|"; //older single-sample GVCFs don't have a value for NON-REF -- GenomicsDB assigns an empty value, which is fair
+        final VariantContext withNonRefValue = new VariantContextBuilder(GATKVariantContextUtils.makeFromAlleles("good", "chr1", 10001, Arrays.asList("A","T", Allele.NON_REF_STRING))).
+                attribute(GATKVCFConstants.AS_RAW_QUAL_APPROX_KEY, goodQualList).make();
+        final VariantContext withNoNonRefValue = new VariantContextBuilder(GATKVariantContextUtils.makeFromAlleles("good", "chr1", 10001, Arrays.asList("A","T", Allele.NON_REF_STRING))).
+                attribute(GATKVCFConstants.AS_RAW_QUAL_APPROX_KEY, trickyQualList).make();
+        Assert.assertTrue(AS_QualByDepth.parseQualList(withNonRefValue).size() == withNonRefValue.getAlternateAlleles().size());
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/allelespecific/AS_QualByDepthUnitTest.java]: `assertEquals()` would be more appropriate here

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/allelespecific/AS_QualByDepthUnitTest.java [hunk 2]
```diff
@@ -27,4 +34,16 @@ protected String getKey() {
...
+        Assert.assertTrue(AS_QualByDepth.parseQualList(withNoNonRefValue).size() == withNonRefValue.getAlternateAlleles().size());
+    }
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/allelespecific/AS_QualByDepthUnitTest.java]: Can you also add an assertion that the actual QD value assigned to NON_REF was 0 in the `withNoNonRefValue` case?

**Comment by ldgauthier** [comment]
@droazen back to you

**Comment by droazen** [comment]
@ldgauthier Looks like you have a test failure in `AS_QualByDepthUnitTest.testParseQualList` (https://storage.googleapis.com/hellbender-test-logs/build_reports/master_29045.13/tests/test/index.html)

**Comment by ldgauthier** [comment]
@droazen I immediately pushed another commit -- see "Derp" above.  Looks green now.

**Pull request merged**